### ISSUE TITLE
Caustics warping, moving and rotating with camera

### ISSUE THREAD

**Issue by daleeidd** [crest-ocean/crest#647]
**Describe the bug**
Caustics distortion causing caustics to warp, move and rotate with camera.

**Screenshots / video**
https://youtu.be/qCGQK7wareI
https://youtu.be/2Wx3hzjoS6o

**Versions**
Unity 2020.1.4f1 Crest URP V4.5 URP Version 8.2.0

**Platform**
 - Windows and MacOS

**Additional context**
It appears to be the refraction code. If I use the old distortion code which uses the distortion texture, there is no issue. It could be a matrix issue or something in ComputeWorldSpacePosition.

**Comment by daleeidd** [comment]
I'll try and create a minimal reproducible case to submit a bug to Unity. And also a workaround.

**Comment by huwb** [comment]
Ha that's surprising that it works with distortion but not refraction. What's the difference - refraction uses normals?

**Comment by daleeidd** [comment]
It does. The main difference is that the refraction approach computes the scene position with the following:

```
float3 scenePos = ComputeWorldSpacePosition(scenePosNDC, sceneZRefractDevice, UNITY_MATRIX_I_VP);
```

I already had to bypass this for the same reason for XR a while ago. Seems to be an issue for normal displays too now :\

I did find this [Unity bug report for UNITY_MATRIX_I_VP](https://issuetracker.unity3d.com/issues/urp-8-dot-1-breaks-unity-matrix-i-vp), but it doesn't solve the issue (caustics won't work at all before that fix).

**Comment by daleeidd** [comment]
Found the issue:
```
#if UNITY_UV_STARTS_AT_TOP
   scenePosNDC.y = 1. - scenePosNDC.y;
#endif
```

It appears this is no longer needed since URP 8.2 (maybe even 7.4. Will check). I will fix it upstream tomorrow.

**Comment by huwb** [comment]
Nice!

**Comment by daleeidd** [comment]
Fix will be in for Crest 4.6.

If anyone is experiencing the issue in URP 7.4 or higher, open the following file and remove the following lines of code:

_/Crest/Crest/Shaders/OceanEmission.hlsl_

```shader
#if UNITY_UV_STARTS_AT_TOP
   scenePosNDC.y = 1. - scenePosNDC.y;
#endif
```

**Comment by adamjm123** [comment]
Fix looks good, thanks Dale

> Fix will be in for Crest 4.6.
> 
> If anyone is experiencing the issue in URP 7.4 or higher, open the following file and remove the following lines of code:
> 
> _/Crest/Crest/Shaders/OceanEmission.hlsl_
> 
> ```glsl
> #if UNITY_UV_STARTS_AT_TOP
>    scenePosNDC.y = 1. - scenePosNDC.y;
> #endif
> ```

Fix looks good, thanks Dale!

**Comment by daleeidd** [comment]
No worries!

4.6 is out with the fix.
